### PR TITLE
Deposit APT tokens to a round in Deposit Phase

### PR DESCRIPTION
**Pull request by 0xriazaka**

### CODE REVIEW

File: vaults/sources/Vault.move [hunk 1]
```diff
@@ -17,18 +17,27 @@ module titusvaults::Vault {
...
 
     struct VaultMap <phantom VaultT, phantom AssetT> has key {
         deposits: SmartTable<address, u64>,
-        total_vaults: u64,
+        shares: SmartTable<address, u64>,
+        total_vaults: u64
+    }
+
+    // events
+    struct DepositVaultEvent {
```
Comment (codebuster22) [inline vaults/sources/Vault.move:37]: can we add the round ID when user contributed

File: vaults/sources/Vault.move [hunk 2]
```diff
@@ -17,18 +17,27 @@ module titusvaults::Vault {
...
         intial_time: u64,
-        round: u64,
+        round: u64
     }
 
     struct Vault <phantom VaultT, phantom AssetT> has key {
         coin_store: Coin<AssetT>,
         creation_time: u64,
         creation_round: u64,
+        total_shares: u64
```
Comment (codebuster22) [inline vaults/sources/Vault.move:27]: are this total shares in a round or total shares in all rounds combined?

File: vaults/sources/Vault.move [hunk 3]
```diff
@@ -40,13 +49,13 @@ module titusvaults::Vault {
         });
     }
 
-    entry fun update_round(_host: &signer) acquires CurrentRound{
+    entry fun update_round(_host: &signer) acquires CurrentRound {
         let host_addr = address_of(_host);
         assert!(host_addr == @titusvaults, E_NOT_AUTHORIZED);
 
         let current_round_struct = borrow_global<CurrentRound>(@titusvaults);
         while (true) {
```
Comment (codebuster22) [inline vaults/sources/Vault.move]: won't this go under infinite loop?

File: vaults/sources/Vault.move [hunk 4]
```diff
@@ -17,18 +17,27 @@ module titusvaults::Vault {
 
     struct CurrentRound has key {
         intial_time: u64,
-        round: u64,
+        round: u64
     }
 
     struct Vault <phantom VaultT, phantom AssetT> has key {
```
Comment (codebuster22) [inline vaults/sources/Vault.move:23]: now I understand, rename this to Round actually.

**Review by codebuster22 — CHANGES_REQUESTED** [review]
looks good. Few minor changes needed.

File: vaults/sources/Vault.move (hunk 2 above)
Comment (0xriazaka) [reply vaults/sources/Vault.move:27]: all rounds combined

File: vaults/sources/Vault.move (hunk 4 above)
Comment (0xriazaka) [reply vaults/sources/Vault.move:23]: ok

**Pull request closed without merging**